tests/python/relay/aot/test_crt_aot.py

```diff
@@ -348,7 +348,9 @@ def test_byoc_utvm(use_calculated_workspaces):
     mod = tvm.IRModule()
     ann = CcompilerAnnotator()
     mod["main"] = ann.visit(f)
+
     mod = tvm.relay.transform.PartitionGraph()(mod)
+    mod = tvm.relay.transform.MangleGraph("my_mod")(mod)
```

Comment:
Is there a use case, we would not want to name mangle ? Im think name mangled function names are strictly superior. If thats the case, I dont like the idea having to run another pass for all BYOC codegens. 

Seeing the previous conversation with @areusch , I agree that we dont need another PartitionGraphWithModName but also not sure we need to call a seperate pass explicitly either. My suggestion is we should name mangle by default otherwise the compilation pathway feels bit fragile.

At a minimum, we should move this pass as a sub-pass of PartitionGraph and call it by default. 
This comment also has a synergy with why we need LowerWithModName in compile engine. 

I think Im missing whats the value of not performing name mangling when user do provide with a mod_name.